index.ts

```diff
@@ -188,6 +198,39 @@ export function load(app: Application) {
 		const outFile =
 			app.options.getValue("coverageOutputPath") ||
 			join(event.outputDirectory, "coverage.svg");
-		writeFileSync(outFile, badge);
+		const outFileJson = outFile.replace(".svg", ".json");
+		const outputType = app.options.getValue("coverageOutputType") || "svg";
+		switch (outputType) {
+			case "svg":
+				writeFileSync(outFile, badge);
+				break;
+			case "json":
+				writeFileSync(
+					outFileJson,
+					JSON.stringify({
+						percent: percentDocumented,
+						expected: expectedCount,
+						actual: actualCount,
+						notDocumented: notDocumented.map((r) => r.getFullName()),
+					}),
+				);
+				break;
+			case "all":
+				writeFileSync(
+					outFileJson,
+					JSON.stringify({
+						percent: percentDocumented,
+						expected: expectedCount,
+						actual: actualCount,
+						notDocumented: notDocumented.map((r) => r.getFullName()),
+					}),
+				);
+				writeFileSync(outFile, badge);
+				break;
+			default:
```

Comment:
I'd prefer to catch this earlier - TypeDoc accepts a `validate` option on the object passed to `addDeclaration` which can be used to reject the invalid value when reading options, rather than potentially several minutes later during rendering

Reply:
Fixed: coverageOutputType is now being checked when TypeDoc validates all options.